plugin/trino-oracle/src/test/java/io/trino/plugin/oracle/BaseOracleCastPushdownTest.java

```diff
@@ -0,0 +1,382 @@
+                .isNotFullyPushedDown(ProjectNode.class);
+    }
+
+    @Override
+    protected List<CastTestCase> supportedCastTypePushdown()
+    {
+        return ImmutableList.of(
+                new CastTestCase("c_char_10", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_char_50", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_char_501", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_char_520", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_nchar_10", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_varchar_10", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_varchar_10_byte", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_varchar_1001", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_varchar_1020", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_nvarchar_100", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_clob", "char(50)", Optional.of("c_char_50")),
+                new CastTestCase("c_nclob", "char(50)", Optional.of("c_char_50")),
+
+                new CastTestCase("c_char_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_char_50", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_char_501", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_char_520", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nchar_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_varchar_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_varchar_10_byte", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_varchar_1001", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_varchar_1020", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nvarchar_100", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_clob", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_nclob", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_3", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_5", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_10", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_19", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_15", "varchar(50)", Optional.of("c_varchar_50")),
+                new CastTestCase("c_number_10_2", "varchar(50)", Optional.of("c_varchar_50")),
```

Comment:
In test methods except of pushdown behaviour we are interested in asserting actual result too.

Reply:
Won't the test will be crowded? I think asserting pushdown also verify the result as well.

Reply:
Right, `Verifies query is fully pushed down and that results are the same as when pushdown is fully disabled.`